packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -259,8 +259,10 @@ export class PureDataObjectFactory<
 		runtime: IContainerRuntime,
 		initialState?: I["InitialState"],
 	): Promise<TObj> {
-		const context = runtime.createDetachedRootDataStore([this.type], rootDataStoreId);
-		return this.createInstanceCore(context, initialState);
+		const dataStore = (runtime as ContainerRuntime).createDataStore2([this.type]);
+		const instance = await this.createInstanceCore(dataStore.context, initialState);
+		await dataStore.trySetAlias(rootDataStoreId);
+		return instance;
```

Comment:
Hmm.. this is tricky. `await dataStore.trySetAlias(rootDataStoreId);` may fail. How do we know if this is the right datastore (the one which got aliased)?

Should this be something like:

```
		const dataStore = (runtime as ContainerRuntime).createDataStore2([this.type]);
		const result = await dataStore.trySetAlias(rootDataStoreId);
        const aliasedDataStore = result === "Success" 
                    ? dataStore
		            : await runtime.getAliasedDataStoreEntryPoint(rootDataStoreId);
		const instance = await this.createInstanceCore(aliasedDataStore.context, initialState);
		return instance;
```

Reply:
Good point. Another option would be to fail creation if an alias exist. That way we give the caller the opportunity to choose a different alias if they want. I suggest this because this API is explicitly about creation so it should probably not return an existing data store?
Both approaches have their pros and cons. Whatever we choose, we should add documentation for it (via comments on this API at least).

Reply:
Yeah, that's a great idea, I'll do that

Reply:
I've updated the test that for `createRootInstance`, it retrieves the winning instance/alias if two creates happen simultaneously.